promlog/log.go

```diff
@@ -61,3 +61,12 @@ func New(al AllowedLevel) log.Logger {
 	l = log.With(l, "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
 	return l
 }
+
+// NewJSON returns a new leveled oklog logger in the JSON format. Each logged line will be annotated
+// with a timestamp. The output always goes to stderr.
+func NewJSON(al AllowedLevel) log.Logger {
```

Comment:
This shouldn't be here. This is something for the user to configure, not the developer.

Reply:
Would you recommend that this function instead exist wherever users within prometheus are setting up their loggers? It made sense to put this in the same place as `New`, since these are doing equivalent things, except with different loggers.

Reply:
This function shouldn't exist at all.